### ISSUE TITLE
Deprecate multi-from

### ISSUE THREAD

**Issue by philrz** [brimdata/zed#5143]
## tl;dr

We intend to deprecate multi-`from` syntax, e.g., users that previously did:

```
from (
  pool sample.zng
  pool prs.json
  get https://ifconfig.co/
)
```

Would now use something like:

```
fork (
  => from sample.zng
  => from prs.json
  => get https://ifconfig.co/
)
```

To give users time to transition, we intend to surface a warning in the Zed tools when a user runs a program that uses the multi-`from` syntax.

## Details

At the time this issue is being opened, Zed is at commit 7c18667.

We've recognized that we could more easily maintain momentum with our recent improvements in error handling if we could drop support for multi-`from`, i.e., this lower portion of what's shown in the [Synopsis portion of the `from` docs](https://zed.brimdata.io/docs/language/operators/from#synopsis):

```
from (
   pool <pool>[@<commitish>] [ => <leg> ]
   pool <pattern>
   file <path> [format <format>] [ => <leg> ]
   get <uri> [format <format>] [ => <leg> ]
   pass
   ...
)
```

While not specific to its use with [`join`](https://zed.brimdata.io/docs/language/operators/join), there was a time that the multi-`from` syntax was the way shown most prominently in the docs for lining up the two `join` inputs. While the main `join` doc no longer shows multi-`from`, we still currently reference it in the [`join` tutorial](https://zed.brimdata.io/docs/tutorials/join) as the ["alternate syntax"](https://zed.brimdata.io/docs/tutorials/join#alternate-syntax). We're aware of a community zync user org that still makes heavy use of multi-`from` with `join`, so their successful transition to the preferred syntax will be important.

We've reached consensus on a multi-part plan:

1. The relevant sections of the Zed docs will be updated to emphasize that the multi-`from` syntax is being deprecated. The syntax we advise using instead will be shown along with a suggestion that the reader ping us on community Slack if they need assistance modifying their programs.

2. The Zed tools will be enhanced to surface a deprecation warning if a user executes a program that uses multi-`from`, including a hyperlink to the docs that provide the explanation and guidance.

3. After we're confident the community zync user org has transitioned and we've given the rest of the community adequate time to react, the multi-`from` syntax will be dropped from the language and treated as an error.

A prerequisite before moving forward is that we may need to think more about the syntax we'll advise. While the `fork` variant shown in the opening issue text above is functional, @mccanne expressed some concern about `fork` as a verb here. When used to split dataflow mid-program it seems to make sense, but it may be seen as an awkward way to kick off a program as users often do today with multi-`from`, i.e., is it intuitive to `fork` nothing?

**Comment by philrz** [comment]
The changes in linked PR #5970 have retired the previous `from` syntax. As of super commit 47b1fb7, the equivalent of the "before" query shown at the top of this issue now looks like:

```
$ super -version
Version: 47b1fb7d8

$ super -s -c "
fork
  ( from sample.bsup )
  ( from prs.json )
  ( from https://ifconfig.co/ )"
```

As for the gradual deprecation spoken of above, the wider changes in the SuperSQL era have been significant enough that we've come to accept that there's going to be a fair number of "breaking changes" we'll be documenting and asking legacy users to adapt to, rather than trying to maintain backward compatibility and/or offer extended periods of support for old/new syntaxes with deprecation guidance. Once we publish our first GA release of SuperDB, if there's sets of users heavily impacted by the loss of old syntax that speak up, we may consider efforts to reintroduce some backward compatibility.
